tests/suites/test_suite_psa_crypto.function

```diff
@@ -3152,6 +3591,574 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void aead_multipart_encrypt( int key_type_arg, data_t *key_data,
+                             int alg_arg,
+                             data_t *nonce,
+                             data_t *additional_data,
+                             int test_ad_mp_arg,
+                             data_t *input_data,
+                             int test_data_mp_arg,
+                             int test_set_lengths_arg,
+                             data_t *expected_result_arg )
+{
+    size_t ad_part_len = 0;
+    size_t data_part_len = 0;
+
+    if( test_ad_mp_arg == 1 )
+    {
+        for( ad_part_len = 1; ad_part_len <= additional_data->len;
```

Comment:
This would, in the current version cause an infinate loop. My current thinking on this was to have a loop alternating zero and ad_part_len (same for data) to check the zero case.